core/src/test/java/io/confluent/rest/DosFilterTest.java

```diff
@@ -48,23 +55,142 @@ public void dosFilterEnabled_throttlesRequests() throws Exception {
     Server server = application.createServer();
     server.start();
 
-    Client client = ClientBuilder.newClient();
+    HttpGet request =
+        new HttpGet(UriBuilder.fromUri(server.getURI()).path("/foo").build());
+
+    CloseableHttpClient ephemeralClient =
+        HttpClients.custom()
+            .setConnectionManager(new BasicHttpClientConnectionManager())
+            .setKeepAliveStrategy((httpResponse, httpContext) -> -1)
+            .build();
 
     // Request should succeed.
-    Response response1 = client.target(server.getURI()).path("/foo").request().get();
-    assertEquals(Status.OK.getStatusCode(), response1.getStatus());
+    CloseableHttpResponse response1 = ephemeralClient.execute(request);
+    assertEquals(Status.OK.getStatusCode(), response1.getStatusLine().getStatusCode());
+    response1.close();
 
     // Following requests should all be throttled.
     for (int i = 0; i < 100; i++) {
-      Response response2 = client.target(server.getURI()).path("/foo").request().get();
-      assertEquals(Status.TOO_MANY_REQUESTS.getStatusCode(), response2.getStatus());
+      CloseableHttpResponse response2 = ephemeralClient.execute(request);
+      assertEquals(
+          Status.TOO_MANY_REQUESTS.getStatusCode(), response2.getStatusLine().getStatusCode());
+      response2.close();
+    }
```

Comment:
I'm a bit concerned about my previous suggestion to bump up the number of requests, but for now we haven't seen flakiness here...

Reply:
Acknowledged.